### PR TITLE
Updates to Web Import Library and CoreLib

### PR DESCRIPTION
**Pull request by mattclay**
I've added some missing properties and methods to the History, Location and Navigator objects, as well as adding XML doc comments to some of the existing ones.

I also added more Func<> and Action<> delegates to CoreLib (3 to 8 args).

### CODE REVIEW

**Comment by nikhilk** [comment]
Originally there was a goal to provide some level of abstraction by using techniques available in other browsers to add IE-specific APIs for very common scenarios. As you mention this is what sscompat.js was doing.

However, this never fully worked, and more importantly, was never accepted (when we did this in the context of microsoft ajax, which is where the technique originated)... folks wanted to code to standard APIs and not IE-specific APIs, and when that wasn't possible, code to an abstraction that surfaced a new api and worked consistently in every browser with different implementation underneath as needed (ala jQuery).

So here is my current thought:
1. The script# compiler doesn't do any browser abstraction.
2. Script# doesn't provide an abstraction/compat script - primarily because jQuery does the job and is widely accepted where folks do need to target the range of browsers. Just delegate to it at this point, or if someone doesn't want all of jQuery, they can implement some script that normalizes the specific APIs they care about and define an imported API. Also, the abstraction/normalization bit changes over time ... for example, the approach taken by modernizer which is in some ways similar to a compat script is also getting some attention.
3. I split out the HTML APIs out of mscorlib.dll into Script.Web.dll so it could enable creation of a Script.Web.Html5.dll at some point that contained only standard APIs (available in modern browsers - IE9+, and webkit-based browsers) and a Script.Web.dll that wasn't so clean.
4. Create a metadata attribute to indicate APIs that are only present in some browsers, and then an FxCop rule that flagged usage of APIs not available in the intended set of target browsers for a particular script.

So in short, I think the value script# could offer is in providing and enabling static analysis tools that help build cross-browser solutions, rather than solve the abstraction problem because that is solved by various frameworks.

**Comment by robitar** [comment]
Makes sense
